feat: refactor web-component in external library

cristianvasquez opened the pull request:
Currently, this web component uses server-side rendering. 

I refactored the web component part into a new library [zazuko/rdf-entity-webcomponent](https://github.com/zazuko/rdf-entity-webcomponent)

That can be used in web environments as well, without node. 

This pull request is to import the new lib and clean up the repo.

cristianvasquez committed:
feat: refactor web-component in external library

cristianvasquez committed:
feat: update web-component-lib
feat: add showNamedGraphs param

cristianvasquez committed:
feat: update web-component-lib
feat: add showNamedGraphs param

cristianvasquez committed:
Merge remote-tracking branch 'origin/use-web-component' into use-web-component

# Conflicts:
#	renderer/clownface.js

ludovicm67 commented on examples/static/welcome.hbs:
```diff
@@ -36,6 +36,10 @@
-
+        <h2>Show named graphs</h2>
+        <ul>
+            <li><a href="http://localhost:3000/adams/eudora?showNamedGraphs=true">Eudora (debug=true)</a></li>
```
I think that you may want to change `(debug=true)` to `(showNamedGraphs=true)`?

ludovicm67 commented on package.json:
```diff
@@ -28,17 +27,11 @@
-    "lit": "^2.3.1",
-    "rdf-ext": "^2.0.1"
+    "rdf-ext": "^2.0.1",
+    "rdf-entity-webcomponent": "^0.2.0"
```
Can we make sure that in the future we publish new packages in the `@zazuko/` namespace?

cristianvasquez replied:
Can one use the namespace for specific things also?

for example `@zazuko/trifid`?

ludovicm67 replied:
For Trifid, either `@zazuko/trifid` or `@trifid/trifid`.
This is something that we might need to discuss at some point.

ludovicm67 reviewed (changes requested):
Yes, good idea! I think it can make sense to have all that webcomponent work properly outside this repo!

For future packages, I think that we really need to publish packages under the `@zazuko/` prefix (just have the name of the module starting with `@zazuko/`). Can you make sure that it will be the case for the next packages that you will create in the future?
This will be helpful for us as a company to manage the access.

When I'm starting the app, I'm getting the following warning:

```
The main 'lit-element' module entrypoint is deprecated. Please update your imports to use the 'lit' package: 'lit' and 'lit/decorators.ts' or import from 'lit-element/lit-element.ts'. See https://lit.dev/msg/deprecated-import-path for more information.
```

Can you investigate on having a fix for it?

Regarding the name of some options, I will have some suggestions:
- Embed blanks: I guess that this is for enabling the embedding of the blank nodes (at least it is what I'm able to see when I try the example instance). To have it more clear, I suggest having `embedBlankNodes` instead of. `embedBlanks`,
- Highlight language: for this, I will suggest renaming the option `highLightLanguage` to `highlightLanguage`, since `highlight` is a single word.

For the "Highlight language" option, is it possible to have a bit more contrast?
It's hard to visualize the difference, because colors looks very similar.
Also on the example instance, you have the "Eudora (highLightLanguage=es)", can you pass the setting also in the URL for `es` as you have for the `fr` link?

ludovicm67 commented:
I also saw that on the example instance some items are not rendered in HTML, like this one: http://localhost:3000/adams/wednesday. Do you have an idea why?

cristianvasquez commented:
> I also saw that on the example instance some items are not rendered in HTML, like this one: http://localhost:3000/adams/wednesday. Do you have an idea why?

yes, I have for demo purposes some strict rules for each of the adams. The ones that are 'not configured' are not shown.

```yaml
  embedNamed:
    paths: /adams/gomez
    module: "file:../../index.js"
    config:
      embedBlanks: true

  noTechnicalCues:
    paths: /adams/morticia
    module: "file:../../index.js"
    config:
      technicalCues: false

  compactMode:
    paths: /adams/uncle
    module: "file:../../index.js"
    config:
      compactMode: false

  highLightLanguage:
    paths: /adams/eudora
    module: "file:../../index.js"
    config:
      compactMode: true
      highLightLanguage: 'es'
```

Probably w could deprecate this when the demo instance is up

cristianvasquez committed:
chore:rdf-entity-webcomponent
feat: add showNamedGraphs param

cristianvasquez commented:
@ludovicm67 I evaluated changing to embedBlankNodes instead of embedBlanks, but I found that in the WebComponent the properties embed-named-nodes embed-blank-nodes don't work because reasons I ignore :/ .

```html
<rdf-entity
            named-graphs
            compact-mode
            embed-named
            !embed-blanks
            debug></rdf-entity>
```

I will keep them in two words in the meantime to keep consistency between the two

cristianvasquez commented:
Can I merge?

ludovicm67 commented:
So why it is working for `highLightLanguage`?

Before merging, here are some points:
- renaming `highLightLanguage` to `highlightLanguage`, since highlight is a single word,
- renaming `embedBlanks` to `embedBlankNodes` (if possible)
- fix the view so that it reflects the change of the renaming of the options
- replace `debug` with `showNamedGraphs`: https://github.com/zazuko/trifid-renderer-entity/blob/2f1347c0b5faaadccbf4c919a15024b7137e667e/examples/static/welcome.hbs#L41-L42
- for the "Highlight language" option, is it possible to have a bit more contrast? It's hard to visualize the difference, because colors looks very similar. Was this point fixed?

ludovicm67 commented:
Regarding the renaming of options, I created a PR here: https://github.com/zazuko/rdf-entity-webcomponent/pull/1
Can you review it and check if it is working as expected @cristianvasquez?

For what I've tested, it seemed to be working as expected.

ludovicm67 committed:
chore: add Zazuko as author

ludovicm67 committed:
ci: use node 18

ludovicm67 committed:
feat(options): rename highLightLanguage to highlightLanguage

ludovicm67 committed:
feat(options): rename embedBlanks to embedBlankNodes

ludovicm67 committed:
style: format welcome view

ludovicm67 committed:
fix: replace debug with showNamedGraphs in view

ludovicm67 committed:
feat(options): rename embedNamed to embedNamedNodes

ludovicm67 committed:
chore(deps): upgrade rdf-entity-webcomponent to 0.5.0

cristianvasquez committed:
chore: make letters smaller when highlight-language

ludovicm67 committed:
style: format welcome view

cristianvasquez committed:
refactor:

embedNamed -> embedNamedNodes
embedBlanks -> embedBlankNodes
highLightLanguage -> highlightLanguage

namedGraphs -> showNamedGraphs

cristianvasquez committed:
Merge remote-tracking branch 'origin/use-web-component' into use-web-component

# Conflicts:
#	examples/config/adams.yaml
#	examples/static/welcome.hbs
#	package.json
#	renderer/clownface.js

ludovicm67 reviewed (approved):
LGTM, good work @cristianvasquez!
